### PR TITLE
D8MT-2399: Code upgraded to provide compatibility with Drupal 9.

### PR DESCRIPTION
**Pull request by emhernapa**
### Description

Based on the results performed in the analysis for the Drupal 9 code compatibility, perform the necessary code refactoring in order to ensure that "oe_migration" code is compatible with Drupal 9.

### Change log

- Changed: Provided code compatibility with Drupal 9.

### CODE REVIEW

**Review by MiguelArber — DISMISSED** [review]
Drupal 9 compatibility changes seem to be correct.

File: modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php [hunk 1]
```diff
@@ -205,19 +205,6 @@ public function testInvalidArgumentConfiguration(array $config, $message) {
     $this->initializePlugin($config);
   }
 
-  /**
-   * Test with an invalid workflow name.
-   */
-  public function testInvalidWorkflowName() {
```
Comment (upchuk) [inline modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php]: Why was this test removed?

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php (hunk 1 above)
Comment (MiguelArber) [reply modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php]: Hi @upchuk, thanks for noticing. It was removed temporarily and I didn’t restore it. I will put it back again, thanks.

Comment (MiguelArber) [reply modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php]: Missing test was restored.

File: modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php [hunk 2]
```diff
@@ -205,6 +205,19 @@ public function testInvalidArgumentConfiguration(array $config, $message) {
     $this->initializePlugin($config);
   }
 
+  /**
```
Comment (MiguelArber) [inline modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php:208]: @upchuk Test was restored. Thank you.
Cc.: @ioannis-ampatzis

File: modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php [hunk 3]
```diff
@@ -205,6 +205,19 @@ public function testInvalidArgumentConfiguration(array $config, $message) {
...
+  public function testInvalidWorkflowName() {
+    $invalid_workflow_name = $this->getRandomGenerator()->string();
+
+    $configuration = ['workflow_config_name' => $invalid_workflow_name];
+
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage(sprintf('"%s" is not a valid workflow.', $invalid_workflow_name));
+    $this->initializePlugin($configuration);
+  }
+
```
Comment (MiguelArber) [inline modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php:220]: @upchuk Missing test was restored. Thanks for your feedback.
Cc.: @ioannis-ampatzis

**Review by ioannis-ampatzis — APPROVED** [review]
After the restoration of the deleted test, everything works as expected.

**Review by upchuk — APPROVED** [review]

**Pull request merged**